src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,623 @@
+                SyntaxFactory.Parameter(
+                    new SyntaxList<AttributeListSyntax>(p.AttributeLists.SelectAsArray(attributeList =>
+                        attributeList.Target == null
+                        // convert attributes attached to the property with no target into "property :" targeted attributes
+                        ? attributeList
+                            .WithTarget(SyntaxFactory.AttributeTargetSpecifier(SyntaxFactory.Token(SyntaxKind.PropertyKeyword)))
+                            .WithoutTrivia()
+                        : attributeList.WithoutTrivia())),
+                    modifiers: default,
+                    p.Type,
+                    p.Identifier,
+                    @default: null));
+
+            // remove converted properties and reformat other methods
+            var membersToKeep = GetModifiedMembersForPositionalRecord(originalType, originalDeclarationNode, semanticModel, propertiesToMove, cancellationToken);
+
+            var changedTypeDeclaration = SyntaxFactory.RecordDeclaration(
+                originalType.TypeKind == TypeKind.Class
+                    ? SyntaxKind.RecordDeclaration
+                    : SyntaxKind.RecordStructDeclaration,
+                originalDeclarationNode.GetAttributeLists(),
+                originalDeclarationNode.GetModifiers(),
+                SyntaxFactory.Token(SyntaxKind.RecordKeyword),
+                originalType.TypeKind == TypeKind.Class
+                    ? default
+                    : SyntaxFactory.Token(SyntaxKind.StructKeyword),
+                // remove trailing trivia from places where we would want to insert the parameter list before a line break
+                originalDeclarationNode.Identifier.WithoutTrailingTrivia(),
+                originalDeclarationNode.TypeParameterList?.WithoutTrailingTrivia(),
+                SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(propertiesToAddAsParams)),
+                // TODO: inheritance
+                baseList: null,
+                originalDeclarationNode.ConstraintClauses,
+                SyntaxFactory.Token(SyntaxKind.OpenBraceToken),
+                SyntaxFactory.List(membersToKeep),
+                SyntaxFactory.Token(SyntaxKind.CloseBraceToken),
+                semicolonToken: default)
+                .WithLeadingTrivia(modifiedClassTrivia)
+                .WithAdditionalAnnotations(Formatter.Annotation);
+
+            var changedDocument = await document.ReplaceNodeAsync(originalDeclarationNode, changedTypeDeclaration, cancellationToken).ConfigureAwait(false);
+            return changedDocument;
+        }
+
+        private static bool ShouldConvertProperty(MemberDeclarationSyntax member, INamedTypeSymbol containingType)
+        {
+            if (member is not PropertyDeclarationSyntax property)
+            {
+                return false;
+            }
+
+            if (property.Initializer != null || property.ExpressionBody != null)
```

Comment:
Is initializer support coming later? perhaps converted to default parameter values?

Reply:
Yes, that's something I could potentially look into. One problem would be that if it had a default value, it would have to be at the end of the parameter list and it might mess up the order of the properties. Currently we don't mess with the order but I guess there's no real reason why we couldn't.

Reply:
also, parameter defaults must be constants, whereas initializers can be arbitrary expressions (like accessing statics).  so this would be complex to support (but nice).

Reply:
I think what would make more sense for parameter defaults is if we find a constructor (maybe one that has the same types as the primary constructor), we move their default parameter values and potentially re-order our own parameters to fit that constructor.